runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7133,6 +7134,31 @@ TR::CompilationInfoPerThreadBase::shouldPerformLocalComp(const TR_MethodToBeComp
   
    return (isMemoryCheapCompilation(bcsz, optLevel) && isCPUCheapCompilation(bcsz, optLevel));
    }
+
+void
+TR::CompilationInfoPerThreadBase::enterPerClientAllocationRegion()
+   {
+   OMR::CriticalSection compilationMonitorLock(_compInfo.getCompilationMonitor());
+   TR_ASSERT(_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER, "must be called on the server"); 
+   static const char* disablePerClientPersistentAllocation = feGetEnv("TR_DisablePerClientPersistentAllocation");
+   if (!disablePerClientPersistentAllocation)
```

Comment:
I would expect this option to have global effect and disable per-client allocation altogether, not just the regions. Was this the intention?

Reply:
Yes, the intention is to completely disable per-client allocation.